src/main/java/com/github/manolo8/darkbot/core/manager/HeroManager.java

```diff
@@ -367,11 +370,13 @@ public Configuration getConfiguration() {
         public void set(ShipMode other) {
             this.configuration = other.getConfiguration();
             this.formation = other.getFormation();
+            this.toSelectChar = null;
```

Comment:
make it depend on `other instanceof Config.ShipMode`, if it is, then set the char to what it has to be, otherwise null